src/schema/posts.ts

```diff
@@ -912,7 +913,19 @@ export const resolvers: IResolvers<any, Context> = {
     checkLinkPreview: async (
       _,
       { url }: SubmitExternalLinkArgs,
-    ): Promise<ExternalLinkPreview> => fetchLinkPreview(url),
+      ctx,
+    ): Promise<ExternalLinkPreview> => {
+      const standardizedUrl = standardizeURL(url);
+      const post = await ctx.con
+        .getRepository(ArticlePost)
+        .createQueryBuilder()
+        .select('id, title, image')
+        .where([{ canonicalUrl: standardizedUrl }, { url: standardizedUrl }])
+        .andWhere({ deleted: false })
+        .getRawOne();
+
+      return post ?? (await fetchLinkPreview(standardizedUrl));
```

Comment:
I actually did that initially ([in this change](https://github.com/dailydotdev/daily-api/pull/1244/commits/641daf960df214de5388f08839a0dbe08847940b#diff-e3ad1f5d3fe340ded80e09a15a95f7d80be472d110fe2bdbbce21dba440b925aR927-R931)), but I thought it would be extremely rare to misunderstand the difference. WDYT?

Reply:
I agree with Ante. If will make it more readable